common/kafka/kafka.go

```diff
@@ -123,27 +112,81 @@ func getTopic(opts map[string][]string, topicType string) (string, error) {
 	return topic, nil
 }
 
-func getCompression(opts map[string][]string) (proto.Compression, error) {
-	if len(opts[compression]) == 0 {
-		return proto.CompressionNone, nil
+func getCompression(opts url.Values) (kafka.CompressionCodec, error) {
+	if len(opts["compression"]) == 0 {
+		return kafka.CompressionNone, nil
 	}
-	comp := opts[compression][0]
+	comp := opts["compression"][0]
 	switch comp {
 	case "none":
-		return proto.CompressionNone, nil
+		return kafka.CompressionNone, nil
 	case "gzip":
-		return proto.CompressionGzip, nil
+		return kafka.CompressionGZIP, nil
+	case "snappy":
+		return kafka.CompressionSnappy, nil
+	case "lz4":
+		return kafka.CompressionLZ4, nil
 	default:
-		return proto.CompressionNone, fmt.Errorf("Compression '%s' is illegal. Use none or gzip", comp)
+		return kafka.CompressionNone, fmt.Errorf("Compression '%s' is illegal. Use none or gzip", comp)
+	}
+}
+
+func getTlsConfiguration(opts url.Values) (*tls.Config, error) {
+	if len(opts["cacert"]) == 0 &&
+		(len(opts["cert"]) == 0 || len(opts["key"]) == 0) {
+		return nil, nil
+	}
+	t := &tls.Config{}
+	if len(opts["cacert"]) != 0 {
+		caFile := opts["cacert"][0]
+		caCert, err := ioutil.ReadFile(caFile)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
It should not use the `log` logger, but the `glog`, but either way this function should have just return with an error. It should not cause call directly `Fatal` because other sink might be functional.

Reply:
Done, sorry it was from my first tests....